### PR TITLE
feat: Adds page layout with header footer and navigation

### PR DESCRIPTION
**Pull request by Ektoplasme**
Adds the page layout:
- Added a `Layout` component containing header, content and footer
- Header made by Pascal, i added a bit of logic and fixed the style
- Footer containing a new component `NavigationBar`, connected to the router so it updates whenever we change the location
- Added a `ConnectedUser` type (in `src/types` folder) to change when we have the real users datas!

To do next:

- [ ] Replace the fake `connectedUser` data in `Layout.tsx` with the real connected user datas (sync with Nicolas that is doing the connexion feature)
- [ ] Update ` ConnectedUser ` type with the correct properties
- [ ] See what we wanna do when we click on the connected user profile in the upleft corner (dropdown menu? or just redirect to the user settings)

https://github.com/user-attachments/assets/d545fbf8-2853-46e8-a7d1-f4eb5cd78adf

If you have any question, do not hesitate to ask

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "@apollo/client": "^3.11.8",
     "@emotion/react": "^11.13.3",
     "@emotion/styled": "^11.13.0",
+    "@mui/icons-material": "^6.1.5",
```
Comment (Ektoplasme) [inline client/package.json:16]: This is needed to use icons in our website

File: client/src/components/Layout/Layout.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { Outlet } from "react-router-dom";
+
+export const Layout = () => {
+  const isMobile = true;
+
+  const connectedUser: ConnectedUser = {
+    username: "Minouche",
+    avatarUrl:
+      "https://imgs.search.brave.com/bWacxyxQbYXhq75ppFyxwnG2z3wK8xDHut72pRbYOZM/rs:fit:860:0:0:0/g:ce/aHR0cHM6Ly9pbWcu/ZnJlZXBpay5jb20v/ZnJlZS1waG90by9h/ZG9yYWJsZS1jYXQt/bGlmZXN0eWxlXzIz/LTIxNTE1OTMzMjEu/anBnP3NlbXQ9YWlz/X2h5YnJpZA",
+  };
```
Comment (Ektoplasme) [inline client/src/components/Layout/Layout.tsx:12]: Those are fake connected user datas, to be replaces with real ones (sync with Nicolas)

Comment (Pas-nou) [reply client/src/components/Layout/Layout.tsx:12]: C'est pas plus simple de leur donner directement les bons nommages "name" et "profile_picture"?

**Review by Pas-nou — DISMISSED** [review]

File: client/src/components/Layout/Layout.tsx (hunk 2 above)
Comment (Ektoplasme) [reply client/src/components/Layout/Layout.tsx:12]: Si carrément

File: client/index.html [hunk 3]
```diff
@@ -3,6 +3,18 @@
...
+    <link
+      href="https://fonts.googleapis.com/css2?family=Londrina+Solid:wght@100;300;400;900&family=Oxanium:wght@200..800&display=swap"
+      rel="stylesheet"
+    />
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+    <link
+      href="https://fonts.googleapis.com/css2?family=Londrina+Solid:wght@100;300;400;900&family=Oxanium:wght@200..800&display=swap"
+      rel="stylesheet"
+    />
```
Comment (Ektoplasme) [inline client/index.html:17]: Ajout des polices depuis Google fonts

**Review by pjroggy — APPROVED** [review]

**Review by Pas-nou — DISMISSED** [review]

**Review by pjroggy — DISMISSED** [review]

File: client/src/components/Header/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import Logo from "/Logo.svg";
+import Avatar from "@mui/joy/Avatar";
+import "./Header.css";
+import { Link } from "react-router-dom";
+import { ConnectedUser } from "../../types/User";
+
+export const Header = ({ username, avatarUrl }: ConnectedUser) => {
+  return (
+    <section className="header-container">
```
Comment (jujuck) [inline client/src/components/Header/Header.tsx]: Balise nav et non section

File: client/src/components/Layout/Layout.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+  return (
+    <div className="layout-container">
+      <div className="layout-header">
```
Comment (jujuck) [inline client/src/components/Layout/Layout.tsx]: Balise section

File: client/src/components/Layout/Layout.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        <Header {...connectedUser} />
+      </div>
+      <div className="layout-content">
```
Comment (jujuck) [inline client/src/components/Layout/Layout.tsx]: Balise main

File: client/src/components/NavigationBar/NavigationBar.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+const iconStyle = {
+  width: "30px",
+  height: "50px",
+};
+
+export default function ManualNavbar() {
+  const location = useLocation();
+
+  const getSelectedTab = () => {
```
Comment (jujuck) [inline client/src/components/NavigationBar/NavigationBar.tsx]: const activeTab = {
   "/": 0
   "/matches": 1,
   "/likes": 2,
   "/settings": 3
}

File: client/src/components/NavigationBar/NavigationBar.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    if (currentPage === "/settings") return 3;
+
+    return 0;
+  };
+
+  return (
+    <Tabs
+      aria-label="tabs"
+      defaultValue={0}
+      value={getSelectedTab()}
```
Comment (jujuck) [inline client/src/components/NavigationBar/NavigationBar.tsx]: value={activeTab[location.pathName]}

**Review by jujuck — DISMISSED** [review]

**Review by Pas-nou — DISMISSED** [review]
Good

**Review by pjroggy — DISMISSED** [review]

File: client/src/components/Layout/Layout.css [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    min-height: 100vh;
+}
+
+.layout-header {
+    z-index: 1;
+    position: sticky;
+    top: 0;
+    height: 80px;
+    width: 100%;
+
```
Comment (jujuck) [inline client/src/components/Layout/Layout.css]: Ligne morte

File: client/src/components/Layout/Layout.css [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    margin-top: auto;
+    margin-bottom: 20px;
+}
+
+.layout-content {
+    max-width: 1280px;
+
```
Comment (jujuck) [inline client/src/components/Layout/Layout.css]: Ligne morte

**Review by jujuck — CHANGES_REQUESTED** [review]

**Review by pjroggy — APPROVED** [review]

**Review by Pas-nou — APPROVED** [review]
Purrfect 👍

**Review by NicolasSokolowski — APPROVED** [review]

**Review by jujuck — APPROVED** [review]

**Pull request merged**